### ISSUE TITLE
Question: Map on FormatBlock

### ISSUE THREAD

**Issue by cornerman** [outr/scribe#115]
What happened to the former `map` function on `FormatBlock`? I found it quite useful for changing the output of some formatted string. What is the alternative?

**Comment by darkfrog26** [comment]
It got removed when `FormatBlock` changed from using `String` to using `LogOutput`.  It would be trivial to add it back with `def map(f: LogOutput => LogOutput): FormatBlock`, but the reason I didn't is that I figure it wouldn't be very useful anymore.

**Comment by cornerman** [comment]
Maybe it is not needed anymore, I am totally open to do it in a different way, I just find `map` to be a quite intuitive method to do transformations on the resulting output.

My use-cases:

```scala
  val shortThreadName = threadName.map(_.replaceFirst("server-akka.actor.default-dispatcher-", "akka-"))
  val shortLevel = level.map(_.trim)
  val fileBaseName = FormatBlock.FileName.map(fileName => fileName.split('/').last)
```

Is there a better way to do it? :)

**Comment by darkfrog26** [comment]
I just added map back in (https://github.com/outr/scribe/commit/87424c0b826221c69e081cc89a740def72b75ee9) if you want publishLocal and test with that, I'd be happy to do a new release if that gives you what you need.

**Comment by cornerman** [comment]
Thank you! Though, I am not sure about the usability as is. As far as I understand `LogOutput`, there are different kinds of `LogOutput` like `ColoredOutput`. Then, we would later pattern match on the `LogOutput` and do some variations of outputting the plaintext depending on the type -- e.g. adding colors for ColoredOutput.

Therefore, I think we would need something like `map(f: String => String)` or `copy(plaintext = _)` on the `LogOutput`, so I can keep the `LogOutput` type as it and just maniuplate the plaintext. This would then lead to code like this: `level.map(_.map(_.trim))`. It might make sense to add back the old `map(f: String => String)` in that case. What do you think?

**Comment by darkfrog26** [comment]
I just pushed a new update that adds `FormatBlock.mapPlain` that I think will do what you want.  Let me know if that does what you need.

**Comment by cornerman** [comment]
Great! That is exactly what I need :)

Thank you so much for all the work you are doing here. This library is awesome and your support is lightning fast!

**Comment by darkfrog26** [comment]
Just released 2.7.9 with this update.  Thanks for the kinda remarks. :)
